### PR TITLE
Modify S3333: Change text to LaYC (APPSEC-1221)

### PR DESCRIPTION
**Pull request by gaetan-ferry-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S3333/php/rule.adoc [hunk 1]
```diff
@@ -1,42 +1,74 @@
...
 
+The main PHP configuration should define the `open_basedir` setting. Moreover,
+this setting should not include overly large directories, such as the root
+directory of the filesystem.
 
-=== Noncompliant code example
+Additionally, adding the current directory, noted ".", to the `open_basedir`
+configuration is also dangerous. Indeed, it is possible to change the current
+directories from PHP scripts with `chdir()`, making the hardening configuration
+useless.
```
Comment (jamie-anderson-sonarsource) [inline rules/S3333/php/rule.adoc:31]:
```suggestion
Adding the current directory, denoted by "`.`", to the `open_basedir`
configuration is also dangerous. It is possible to change the current directory
within PHP scripts by calling `chdir()`, effectively removing any protection.
```

**Review by jamie-anderson-sonarsource — CHANGES_REQUESTED** [review]
There are some bits of unneeded wording at the start of sentences that should be removed.

File: rules/S3333/php/rule.adoc [hunk 2]
```diff
@@ -1,42 +1,74 @@
...
 
-``++open_basedir++`` should be configured with a directory, which will then be accessible recursively. However, the use of ``++.++`` (current directory) as an ``++open_basedir++`` value should be avoided since it's resolved dynamically during script execution, so a ``++chdir('/')++`` command could lay the whole server open to the script.
+While deactivating the `open_basedir` setting does not pose a direct threat to
+the application's security, it can make the exploitation of other
+vulnerabilities easier and more severe.
 
+More precisely, exploiting a path traversal issue in that case will allow
+attackers to access any file on the filesystem without additional restrictions
+to those applied by the operating system permission checks. This includes
+system-critical or otherwise sensitive files.
```
[the remark this answers is not in the document]
Comment (gaetan-ferry-sonarsource) [reply rules/S3333/php/rule.adoc]: I tried to come up with another way of saying it.

**Review by jamie-anderson-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**